zipline/data/bundles/quandl.py

```diff
@@ -13,265 +13,165 @@
 from six.moves.urllib.parse import urlencode
 
 from zipline.utils.calendars import register_calendar_alias
-from zipline.utils.cli import maybe_show_progress
-
+from zipline.utils.deprecate import deprecated
 from . import core as bundles
+import numpy as np
 
 log = Logger(__name__)
-seconds_per_call = (pd.Timedelta('10 minutes') / 2000).total_seconds()
-# Invalid symbols that quandl has had in its metadata:
-excluded_symbols = frozenset({'TEST123456789'})
-
-
-def _fetch_raw_metadata(api_key, cache, retries, environ):
-    """Generator that yields each page of data from the metadata endpoint
-    as a dataframe.
-    """
-    for page_number in count(1):
-        key = 'metadata-page-%d' % page_number
-        try:
-            raw = cache[key]
-        except KeyError:
-            for _ in range(retries):
-                try:
-                    raw = pd.read_csv(
-                        format_metadata_url(api_key, page_number),
-                        parse_dates=[
-                            'oldest_available_date',
-                            'newest_available_date',
-                        ],
-                        usecols=[
-                            'dataset_code',
-                            'name',
-                            'oldest_available_date',
-                            'newest_available_date',
-                        ],
-                    )
-                    break
-                except ValueError:
-                    # when we are past the last page we will get a value
-                    # error because there will be no columns
-                    raw = pd.DataFrame([])
-                    break
-                except Exception:
-                    pass
-            else:
-                raise ValueError(
-                    'Failed to download metadata page %d after %d'
-                    ' attempts.' % (page_number, retries),
-                )
-
-            cache[key] = raw
+warnings.simplefilter('once', DeprecationWarning)
```

Comment:
Does this mean that all DeprecationWarnings from any module will only print once? I don't think we want to decide that for users - I think we probably don't want to set up any filters for any modules. What do you think?

Reply:
I think I misunderstood the correct use of the module on my first pass through the documentation. I found a good explanation [here](http://www.lesinskis.com/python_deprecation_tutorial.html).